lib/features/children/generosity_challenge/assignments/family_values/repositories/family_values_repository.dart

```diff
@@ -0,0 +1,38 @@
+import 'dart:convert';
+
+import 'package:givt_app/features/children/generosity_challenge/assignments/family_values/cubit/family_values_cubit.dart';
+import 'package:givt_app/features/children/generosity_challenge/assignments/family_values/models/family_value.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+mixin FamilyValuesRepository {
+  Future<bool> rememberValues({
+    required List<Map<String, dynamic>> body,
```

Comment:
why do we use map here instead of List<FamilyValue>?

Reply:
Its of list of maps because its a list of family values, i could wrap the list in a map to be something like:
```
{"values" : [
    { ...
    },
    { ...
    },
  ]
}
```
But i am not sure what would be the utility of that?

Reply:
'Its of list of maps because its a list of family values' - yes, and that's ok. I think repository is a good place for _data translation_ from model to map and to json string and backward instead of doing it in the cubit

Reply:
I mean this part:
`final selectedValuesJson =
        state.selectedValues.map((e) => e.toMap()).toList();`

Reply:
Ooh! Yes totally, thank you for the direction 🙏